index.js

```diff
@@ -37,6 +38,9 @@ module.exports = function (url, options) {
       // rewrite url if our request had to follow redirects to resolve the
       // final link destination (for example: links shortened by bit.ly)
       if (response.request.uri.href) url = response.request.uri.href
+      if (opts.decoder) {
+        body = opts.decoder(body)
+      }
```

Comment:
If I'm reading this correctly, the decoder would need to accept the buffer returned from the `request` module and convert it with `opts.decoder` before parsing the metadata. Something like one of the following packages might work(?): 

https://www.npmjs.com/package/encoding-japanese
https://www.npmjs.com/package/iconv-lite

Reply:
```typescript
import detectCharacterEncoding from 'detect-character-encoding'
import iconv from 'iconv-lite'

const decoder = val => {
  const result = detectCharacterEncoding(val)
  if (!result) {
    return ''
  }
  const { encoding } = result
  if (encoding === 'UTF-8') {
    return val.toString()
  }
  if (!iconv.encodingExists(encoding)) {
    return ''
  }
  return iconv.decode(val, encoding)
}
```

Reply:
Ok perfect, that was what i needed. Will try this tomorrow.